pkg/check/ingress.go

```diff
@@ -114,3 +117,151 @@ func ingressDNS() func(ctx context.Context, node *graph.Node, graph *graph.Graph
 		return false, nil, nil
 	}
 }
+
+func ingressExternalDnsTxtOwner() func(ctx context.Context, node *graph.Node, graph *graph.Graph) (bool, []string, error) {
+	conf, err := dns.ClientConfigFromFile("/etc/resolv.conf")
+	if err != nil {
+		conf = &dns.ClientConfig{
+			Servers: []string{
+				"1.1.1.1",
+			},
+			Port: "53",
+		}
+	}
+	dnsClient := &dns.Client{
+		Net: "tcp",
+	}
+	dnsConn, err := dnsClient.Dial(net.JoinHostPort(conf.Servers[0], conf.Port))
+	if err != nil {
+		return func(ctx context.Context, node *graph.Node, graph *graph.Graph) (bool, []string, error) {
+			return true, []string{fmt.Sprintf("unable to connect to dns: %v", err)}, nil
+		}
+	}
+
+	lookupFn := func(host string) ([]string, error) {
+		msg := &dns.Msg{}
+		queryHost := host
+		if !strings.HasSuffix(host, ".") {
+			queryHost = fmt.Sprintf("%s.", host)
+		}
+		msg.SetQuestion(queryHost, dns.TypeTXT)
+
+		result, _, err := dnsClient.ExchangeWithConn(msg, dnsConn)
+		if err != nil {
+			return nil, err
+		}
+
+		if result.Rcode != dns.RcodeSuccess {
+			return nil, fmt.Errorf("response code not success for lookup of %s: %d", host, result.Rcode)
+		}
+
+		results := []string{}
+		for _, ip := range result.Answer {
+			record, ok := ip.(*dns.TXT)
+			if ok {
+				results = append(results, record.Txt...)
+			}
+		}
+
+		if len(results) == 0 {
+			return nil, fmt.Errorf("received no results from lookup of %s", host)
+		}
+
+		return results, nil
+	}
+
+	mu := &sync.Mutex{}
+	var externalDnsTxtOwner *string
+	var externalDnsTxtOwnerErr error
+	toStringPtr := func(s string) *string { return &s }
+	getExternalDnsTxtOwner := func(g *graph.Graph) (string, error) {
+		mu.Lock()
+		defer mu.Unlock()
+
+		if externalDnsTxtOwner != nil {
+			return *externalDnsTxtOwner, externalDnsTxtOwnerErr
+		}
+
+		var externalDns *appsv1.Deployment
+		err := g.Iterate(func(node *graph.Node) error {
```

Comment:
Not sure that this is the right way to go about this as it is going to be really inefficient. Lets hold off on this until we build a simpler rule that needs to do object lookups. This will probably force changing some of the structure of the actual graph data.

Reply:
@phillebaba do you want to wait with merging this PR or do you mean hold off as in we should merge but fix the inefficient handling at a later stage?